lib/hubspot/config.rb

```diff
@@ -14,7 +18,18 @@ def configure(config)
         @hapikey = config["hapikey"]
         @base_url = config["base_url"] || "https://api.hubapi.com"
         @portal_id = config["portal_id"]
-        @logger = config['logger'] || DEFAULT_LOGGER
+        @logger = config["logger"] || DEFAULT_LOGGER
+        @access_token = config["access_token"]
+        @client_id = config["client_id"] if config["client_id"].present?
+        @client_secret = config["client_secret"] if config["client_secret"].present?
+        @redirect_uri = config["redirect_uri"] if config["redirect_uri"].present?
+
+        unless access_token.present? ^ hapikey.present?
```

Comment:
@tmac yeah I'm happy to switch that to not use the XOR operator. But I think logically, the semantics of XOR is correct. If you have both `access_token` and `hapikey`, then calls will not work as expected. You must have exactly one of them, but not both, and not neither. So I could switch it to maybe...
 ```ruby
if (access_token && hapikey) || !(access_token || hapikey)
  fail "You must provide..." 
end
```